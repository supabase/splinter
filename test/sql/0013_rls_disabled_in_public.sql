begin;
  set local search_path = '';
  set local pgrst.db_schemas = 'public';

  -- 0 issues
  select * from lint."0013_rls_disabled_in_public";

  create table public.foo( it int primary key );

  -- 1 issue
  select * from lint."0013_rls_disabled_in_public";

  -- Resolve the issue by enabling RLS
  alter table public.foo enable row level security;

  -- 0 issue
  select * from lint."0013_rls_disabled_in_public";

  -- Confirm that the lint only applies to `public` tables
  -- by creating a table in a different schema and confirming
  -- that the lint does not fire
  create schema xyz;
  create table xyz.bar( it int primary key );

  -- 0 issue
  select * from lint."0013_rls_disabled_in_public";

rollback;
