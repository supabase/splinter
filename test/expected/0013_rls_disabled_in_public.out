begin;
  set local search_path = '';
  set local pgrst.db_schemas = 'public';
  -- 0 issues
  select * from lint."0013_rls_disabled_in_public";
 name | level | facing | categories | description | detail | remediation | metadata | cache_key 
------+-------+--------+------------+-------------+--------+-------------+----------+-----------
(0 rows)

  create table public.foo( it int primary key );
  -- 1 issue
  select * from lint."0013_rls_disabled_in_public";
          name          | level |  facing  | categories |                                              description                                               |                            detail                             |                                        remediation                                         |                       metadata                       |             cache_key             
------------------------+-------+----------+------------+--------------------------------------------------------------------------------------------------------+---------------------------------------------------------------+--------------------------------------------------------------------------------------------+------------------------------------------------------+-----------------------------------
 rls_disabled_in_public | ERROR | EXTERNAL | {SECURITY} | Detects cases where row level security (RLS) has not been enabled on a table in the \`public\` schema. | Table \`public.foo\` is public, but RLS has not been enabled. | https://supabase.com/docs/guides/database/database-linter?lint=0013_rls_disabled_in_public | {"name": "foo", "type": "table", "schema": "public"} | rls_disabled_in_public_public_foo
(1 row)

  -- Resolve the issue by enabling RLS
  alter table public.foo enable row level security;
  -- 0 issue
  select * from lint."0013_rls_disabled_in_public";
 name | level | facing | categories | description | detail | remediation | metadata | cache_key 
------+-------+--------+------------+-------------+--------+-------------+----------+-----------
(0 rows)

  -- Confirm that the lint only applies to `public` tables
  -- by creating a table in a different schema and confirming
  -- that the lint does not fire
  create schema xyz;
  create table xyz.bar( it int primary key );
  -- 0 issue
  select * from lint."0013_rls_disabled_in_public";
 name | level | facing | categories | description | detail | remediation | metadata | cache_key 
------+-------+--------+------------+-------------+--------+-------------+----------+-----------
(0 rows)

rollback;
